src/ucs/debug/log.c

```diff
@@ -296,10 +355,17 @@ ucs_log_default_handler(const char *file, unsigned line, const char *function,
         /* Add source file name to the hash */
         match  = fnmatch(ucs_global_opts.log_component.file_filter, file, 0) !=
                  FNM_NOMATCH;
-        khiter = kh_put(ucs_log_filter, &ucs_log_global_filter, file, &khret);
-        ucs_assert((khret == UCS_KH_PUT_BUCKET_EMPTY) ||
-                   (khret == UCS_KH_PUT_BUCKET_CLEAR));
-        kh_val(&ucs_log_global_filter, khiter) = match;
+
+        filename = ucs_mem_chunk_alloc_str(file);
```

Comment:
why not just use ucs_malloc/ucs_strdup?

Reply:
small allocations that don't need individual tracking, shall we use:
- create: `ucs_strdup()`
- destroy: `kh_foreach_key() { ucs_free() }`, then destroy kh itself

Reply:
Yes

Reply:
fixed, retested final patch, kh foreach needs `const char *` buf `ucs_free` needs unconstified ptr.